src/main/java/com/endava/endabank/controller/UserController.java

```diff
@@ -0,0 +1,89 @@
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.security.Principal;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping(Routes.API_ROUTE + Routes.USERS_ROUTE)
+@AllArgsConstructor
+public class UserController {
+    private UserService userService;
+
+    @PostMapping
+    public Map<String, Object> create(@Valid @RequestBody UserRegisterDto user) {
+        return ResponseEntity.status(HttpStatus.CREATED).body(userService.saveAndSendVerifyEmail(user)).getBody();
+    }
+
+    @GetMapping()
+    @PreAuthorize(Permissions.AUTHORITY_ACCOUNT_VALIDATE)
+    public ResponseEntity<List<UserToApproveAccountDto>> getUsersToApprove() {
+        return ResponseEntity.status(HttpStatus.OK).body(userService.usersToApprove());
+    }
+
+    @PutMapping(Routes.APPROVE_ACCOUNT_ROUTE)
+    @PreAuthorize(Permissions.AUTHORITY_ACCOUNT_VALIDATE)
+    public ResponseEntity<UserToApproveAccountDto> updateUserIsApproved(@PathVariable Integer id, @RequestBody Map<String, Boolean> map) {
+        return ResponseEntity.status(HttpStatus.OK).body(userService.updateUserAccountApprove(id, map.get("value")));
+    }
+
+    @GetMapping(Routes.RESET_PASSWORD_ROUTE + "/{email}")
+    public ResponseEntity<Map<String, Object>> resetPassword(@PathVariable String email) {
+        return ResponseEntity.status(HttpStatus.OK).body(userService.generateResetPassword(email));
+    }
+
+    @PutMapping(Routes.RESET_PASSWORD_ROUTE)
+    public ResponseEntity<Map<String, String>> updateForgotPassword(@RequestBody UpdatePasswordDto updatePasswordDto) throws AccessDeniedException {
+        return ResponseEntity.status(HttpStatus.OK).body(userService.updateForgotPassword(updatePasswordDto));
+    }
+
+    @PutMapping(Routes.UPDATE_PASSWORD)
+    public ResponseEntity<Map<String, String>> updatePassword(Principal principal, @RequestBody UpdatePasswordDto updatePasswordDto) throws AccessDeniedException {
+        UsernamePasswordAuthenticationToken usernamePasswordAuthenticationToken = (UsernamePasswordAuthenticationToken) principal;
+        UserPrincipalSecurity user = (UserPrincipalSecurity) usernamePasswordAuthenticationToken.getPrincipal();
+        return ResponseEntity.status(HttpStatus.OK).body(userService.updatePassword(user, updatePasswordDto));
```

Comment:
|Hi,

What if somethings fail when "userAuthenticationService.logInUser(authentication)" is executed?

I think you should have a try-catch block to handle exceptions and throw custom exceptions instead of technical ones like BadDataException

if you put the returned entity always with ok you are gonna be handling always everything as ok

Please fix it in the same way for every controller class you have and its methods